src/components/TransformWorkflow/FlowEditor/Components/OperationPanel/Forms/CaseWhenOpForm.tsx

```diff
@@ -482,11 +596,13 @@ const CaseWhenOpForm = ({
                       <Controller
                         control={control}
                         name={`clauses.${clauseIndex}.then.col_val`}
-                        rules={{ required: true }}
+                        rules={{ required: advanceFilter === 'no' }}
                         render={({ field }) => (
                           <Autocomplete
                             options={srcColumns}
-                            disabled={advanceFilter === 'yes'}
+                            disabled={
+                              advanceFilter === 'yes' || action === 'view'
```

Comment:
Create a const and use everywhere

const isDisabled= advanceFilter === 'yes' || action === 'view'